### PR TITLE
Declare strict type

### PR DESCRIPTION
**Pull request by nicodh**

### CODE REVIEW

**Comment by nicodh** [comment]
Not sure if there might be cases where someone does not want the declare strict types - so I made it configurable...

**Review by liayn — DISMISSED** [review]

File: Classes/Service/FileGenerator.php [hunk 1]
```diff
@@ -1323,6 +1279,12 @@ public static function getFolderForClassFile($extensionDirectory, $classType, $c
         throw new \Exception('Unexpected classPath:' . $classPath);
     }
 
+    protected function writeClassFile($classFileObject) {
```
Comment (simonschaufi) [inline Classes/Service/FileGenerator.php:1282]: This is not PSR-2 standard formatted

**Review by simonschaufi — CHANGES_REQUESTED** [review]

File: Classes/Parser/NodeFactory.php [hunk 2]
```diff
@@ -120,6 +120,8 @@ public function getFileStatements(File $fileObject)
                 }
                 $stmts = array_merge($stmts, $this->getContainerStatements($namespace));
             }
+        } else {
```
Comment (simonschaufi) [inline Classes/Parser/NodeFactory.php:123]: I would turn this upside down to make it more like "return early if possible":

```php
if (! ...) {
    return $stmts = $this->getContainerStatements($fileObject);
}
foreach(...)
```

**Review by simonschaufi — CHANGES_REQUESTED** [review]

**Comment by simonschaufi** [comment]
Use the Plugin "PHP Inspections (EA Extended)"

**Comment by nicodh** [comment]
Ok - changed it but imho it does not improve readability that much...

**Review by simonschaufi — APPROVED** [review]

**Pull request merged**